### DISCUSSION TITLE
Overlap query

### DISCUSSION THREAD

**Discussion by vvvlad** [jertel/elastalert2#497] [Q&A] [answered]
Hello,
Is it possible in some way to overlap on previously seen records in the query?

Explanation:
Lets say I would like the rule to run every 10 minutes, but look at all the records 1 hour back.
Example:
At 2 PM the rule will run and take all records since 1 PM
At 2:10 PM the rule will run and take ALL the records since 1:10 PM
At 2:20 PM the rule will run and take ALL the records since 1:20 PM
At 2:30 PM the rule will run and take ALL the records since 1:30 PM

And so on...

**Comment by jertel** [comment, marked as answer]
`buffer_time` (https://elastalert2.readthedocs.io/en/latest/running_elastalert.html?highlight=buffer_time#downloading-and-configuring) and `scan_entire_timeframe` (https://elastalert2.readthedocs.io/en/latest/ruletypes.html#scan-entire-timeframe) might be of use for you. However, ElastAlert 2 will remember events it already found during a previous scan so if you're wanting to send repeating alerts based on the same document IDs then that might not be possible.

**Reply by vvvlad** [reply]
Hi @jertel ,
Thanks for the reply!

Your last sentence summarizes the whole question. So basically, it is not possible to go over the same documents more than once...

**Reply by nsano-rururu** [reply]
This may be relevant.
https://elastalert2.readthedocs.io/en/latest/ruletypes.html#aggregation
